onnx/version_converter/adapters/broadcast_forward_compatibility.h

```diff
@@ -0,0 +1,66 @@
+// Adapter for broadcasting ops in default domain from version 6 to 7
+
+#pragma once
+
+#include "onnx/version_converter/adapters/adapter.h"
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+class BroadcastForwardCompatibility final : public Adapter {
+  public:
+    explicit BroadcastForwardCompatibility(const std::string& op_name, const OpSetID&
+      initial, const OpSetID& target): Adapter(op_name, initial, target) {}
+
+    void adapt_broadcast_forward_compatibility(std::shared_ptr<Graph> graph, Node* node)
+      const {
+      // Remove axis and broadcast attributes
+      // Assess whether axis requires reshaping
+      if (node->hasAttribute(kbroadcast)) {
+        const ArrayRef<Value*>& inputs = node->inputs();
+        ONNX_ASSERTM(inputs.size() == 2, "%s in opset version 6 can only broadcast"
+          " between 2 inputs", name().c_str());
+        ONNX_ASSERTM(inputs[0]->has_sizes(), "Shape of first input not available.");
+        std::vector<Dimension> A_sizes = inputs[0]->sizes();
+        ONNX_ASSERTM(inputs[1]->has_sizes(), "Shape of second input not available.");
+        std::vector<Dimension> B_sizes = inputs[1]->sizes();
+        // Also assert that broadcasting syntax are correct if axis is not present
```

Comment:
Probably add another assertion here: check all the dimensions are dims, not params.

https://github.com/onnx/onnx/blob/36517e6bac9c0f0cbf9390c97c39b8106dd4d33e/onnx/common/ir.h#L75